### PR TITLE
Initial Python 3 support.

### PR DESCRIPTION
**Pull request by eisensheng**
Provide initial Python 3 support by using six where there are differences in the code between Python2 and Python3 behavior.

### CODE REVIEW

File: requirements.in [hunk 1]
```diff
@@ -1,7 +1,9 @@
--e git://github.com/mbertheau/python-emarsys@master#egg=python-emarsys
+-e git://github.com/mbertheau/python-emarsys@0.3#egg=python-emarsys
+
 # lowest supported django version
-Django<1.8
+Django>1.8
```
Comment (mbertheau) [inline requirements.in]: Can you keep it compatible with Django-1.8?

Comment (eisensheng) [inline requirements.in]: Unfortunately not. Only the Django 1.9 line is compatible with Python 3.5 which is being used in the signup service. Anything below 1.9 throws up during the test run.

```
  File ".../.tox/py35/lib/python3.5/site-packages/django/utils/html_parser.py", line 12, in <module>
    HTMLParseError = _html_parser.HTMLParseError
AttributeError: module 'html.parser' has no attribute 'HTMLParseError'
```

I changed the version requirements though to be a little bit more strict.

Comment (mbertheau) [inline requirements.in]: That seriously decreases the value of the work for us. Django 1.9 is supported until 8 months from now, whereas Django 1.8 is supported until 20 months from now.

Additionally the django documentation states here https://docs.djangoproject.com/en/1.8/releases/1.8/ that Python 3.5 is supported. In fact a problem with HTMLParseError is mentioned here https://code.djangoproject.com/ticket/23763 and has been fixed here https://github.com/django/django/commit/b07aa52e8a8e4c7fdc7265f75ce2e7992e657ae9 and the fix should be included in the latest 1.8 release. Please check that again.

Is there anything else preventing us from using Django 1.8?

Comment (eisensheng) [inline requirements.in]: Interesting. I was pretty sure that I tested it with the 1.8 line of Django and it didn't work either. I've explicitly pinned the 1.8.14 version and upwards together with 1.9.8 and upwards until 1.10 line for now and it works magically. :sparkles: 

The test suite will be run by tox for the Django 1.8 line and 1.9 line under Python 2.7 and 3.5 now. It was necessary to split the requirements to achieve this though. Pip doesn't like it that much when a dependency is specified twice it seems.

Comment (mbertheau) [inline requirements.in]: We can drop Django 1.9 support.

**Comment by Azd325** [comment]
Has already Python 3 support.

**Pull request closed without merging**